lib/middleware/preferences_middleware.dart

```diff
@@ -40,7 +43,13 @@ Middleware<AppState> _createLoadPreferences() {
       }
 
       List<String> subbedFires = prefs.getStringList('subscribedFires') ?? [];
-      List<Fire> fires = store.state.fires;
+      List<Fire> fires = List.from(store.state.fires);
```

Comment:
Why are you forcing a new reference here?

Reply:
I actually wanted to ask about that, but a day passed and i forgot to, if i don't i get a` type 'List<dynamic>' is not a subtype of type 'List<Fire>'`.
Any idea?

Reply:
It should implicitly cast to `List<Fire>` without problem in runtime. What you're doing is redundant anyway. 

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was good.

Reply:
It should implicitly cast to `List<Fire>` without problem in runtime. What you're doing is redundant anyway.

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was looking good.

Reply:
It should implicitly cast to `List<Fire>` without problem at runtime. What you're doing is redundant anyway.

Did you have any issue without it? Just ran through it with `List<Fire> fires = store.state.fires;` and it was looking good.

Reply:
Yes, it throws a  `type 'List<dynamic>' is not a subtype of type 'List<Fire>'` on that line, and the filter is only applied after i refresh the fires or change the filter.
Could you run it in debug mode and verify that you get the same exception on the catch clause?

Reply:
I've already tried and didn't seem to throw anything. I've even added a breakpoint to the line below and the fires were loading great.

What are the steps to reproduce this?

Reply:
Solved in [
55155a6](https://github.com/FogosPT/fogosmobile/pull/74/commits/55155a673360153de2e612d828d0c11e8cb931e9)